samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+
+If the `partition-key` header does not exist, the pipeline execution continues to the *Message body inspection* step.
+
+### Message body inspection
+
+If message headers inspection is unsuccessful with value extraction of `partition-key` header, then the message body is used to determine the partition key value. The `DistributeMessagesBasedOnPayload` behavior determines the partition value using the mapping function provided via the configuration API. The result of the mapping function is stored as the `partition-key` header value. The inspection logic can raise a `PartitionMappingFailedException` exception if the partition key is not a valid partition key for a given endpoint. For a partition key that is not local, the message is forwarded to the appropriate partition. 
+
+The forwarding/processing decision is made in the same way as in the *Header inspection* step.
+
+NOTE: `PartitionMappingFailedException` is considered an [unrecoverable exception](/nservicebus/recoverability/custom-recoverability-policy.md). Whenever such an exception is raised the message that triggered the exception will be moved immediately to the configured error queue.
+
+### Control message forwarding
+
+When an endpoint instance receives a control message representing either the [Subscribe or Unsubscribe intent](/nservicebus/messaging/headers.md#messaging-interaction-headers-nservicebus-messageintent), the message is forwarded to all other partitions.
+
+### Configuration
+
+To enable Receiver Side Distribution for a specific endpoint, two arguments need to be provided:
+- Endpoint discriminators that are based on Service Fabric partition keys 
+- Mapping function per incoming message type that maps incoming messages to a partition key
+
+Use this code to enable Receiver-Side distribution:
+
+snippet: ConfigureReceiverSideDistribution-CandidateVoteCount
+
+In many low-throughput scenarios Receiver Side Distribution might be enough to get started. To achieve better performance, it would be more efficient to route directly to the partitions from the sender with *Sender-Side Distribution*.
+
+## Sender Side Distribution
+
+Receiver Side Distribution addresses forwarding messages that arrive to the incorrect partition. Forwarding received messages introduces some overhead though. To remove the overhead on the receiver side the Sender Side Distribution approach can be used to distribute messages to the correct endpoint instances based on Service Fabric partitioning information.
+
+Sender Side Distribution can be applied to endpoints hosted inside Service Fabric by using the partition information of the stateful services. This is suitable for endpoints hosted inside the cluster that need to send messages to other endpoints hosted in the cluster. For the endpoints hosted outside of the cluster access to Service Fabric APIs is not possible. Instead, partitioned destination endpoints need to be registered with message mapping for commands sent out.
```

Comment:
Changed that. The intention was: If you are outside of the cluster you need to maintain that information separately.